### PR TITLE
Adding GCE node termination handler as an optional addon. 

### PR DESCRIPTION
**Pull request by vishh**
This step is a pre-requisite for auto-deploying that addon in GKE

cc @mikedanese

### CODE REVIEW

File: cluster/gce/addons/node-termination-handler/daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+          hostPath:
+            path: /etc/srv/kubernetes
+      tolerations:
+        # Run regardless of any existing taints.
+        - effect: NoSchedule
+          operator: Exists
+        - effect: NoExecute
+          operator: Exists
+      containers:
+        - image: k8s.gcr.io/gke-node-termination-handler@sha256:e08ca863a547754fa7b75064bdad04f04cbef86c7b0a181ecc7304e747623181
```
Comment (awly) [inline cluster/gce/addons/node-termination-handler/daemonset.yaml]: Can we plumb the image tag through configure-helper somehow?
That way we can change the version without bash k/k changes.

Comment (vishh) [reply cluster/gce/addons/node-termination-handler/daemonset.yaml]: Good point. Done.

**Comment by awly** [comment]
LGTM
@mikedanese can you take a quick look? I'm not super familiar with bash and yaml configs

**Pull request merged**